### PR TITLE
Deps: combined test-shadow deps

### PR DESCRIPTION
**Pull request by ddymko**
## Description
Took all the open test-shadow deps into a single PR. Wanted to consolidate all of the deps for the tests into a single PR.

Note: I am not sure if we want to do these deps individually or group them. Perhaps for core deps we will want to do them individually.

Also, we will want to backport this to v2. once we merge this in I will make a follow up ticket.

## Related Issues
#4579
#4578
#4577
#4576
#4575
#4574
#4571
#4569
#4566

## Testing
CI goes green

## Checklist

 - [ ] I made sure to update `CHANGELOG.md`.

   Remember, the CHANGELOG needs to mention:
    + Any new features
    + Any changes to our included version of Envoy
    + Any non-backward-compatible changes
    + Any deprecations

 - [x] This is unlikely to impact how Ambassador performs at scale.

   Remember, things that might have an impact at scale include:
    + Any significant changes in memory use that might require adjusting the memory limits
    + Any significant changes in CPU use that might require adjusting the CPU limits
    + Anything that might change how many replicas users should use
    + Changes that impact data-plane latency/scalability

 - [ ] My change is adequately tested.

   Remember when considering testing:
    + Your change needs to be specifically covered by tests.
       + Tests need to cover all the states where your change is relevant: for example, if you add a behavior that can be enabled or disabled, you'll need tests that cover the enabled case and tests that cover the disabled case. It's not sufficient just to test with the behavior enabled.
    + You also need to make sure that the _entire area being changed_ has adequate test coverage.
       + If existing tests don't actually cover the entire area being changed, add tests.
       + This applies even for aspects of the area that you're not changing – check the test coverage, and improve it if needed!
    + We should lean on the bulk of code being covered by unit tests, but...
    + ... an end-to-end test should cover the integration points

 - [ ] I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently.

 - [ ] The changes in this PR have been reviewed for security concerns and adherence to security best practices.

### CODE REVIEW

**Comment by haq204** [comment]
To minimize potential impact maybe consolidate only minor and patch bumps and leave major version bumps in separate PRs?

**Comment by ddymko** [comment]
@haq204 I'll work on making a proposal for this. For this, specifically I am ok with lumping it together. I was curious to see if CI would pass successfully.

File: docker/test-shadow/requirements.txt [hunk 1]
```diff
@@ -4,12 +4,12 @@
...
-importlib-metadata==4.8.3  # via click
-itsdangerous==2.0.1       # via flask
-jinja2==2.10.1            # via flask
-markupsafe==2.0.1         # via jinja2
-typing-extensions==4.0.1  # via importlib-metadata
-werkzeug==2.0.2           # via flask
-zipp==3.6.0               # via importlib-metadata
+click==8.1.3              # via flask
+flask==2.2.2              # via -r requirements.in
+importlib-metadata==4.12.0  # via click
```
Comment (haq204) [inline docker/test-shadow/requirements.txt:9]: nit: align the comments

**Review by haq204 — APPROVED** [review]

**Comment by LukeShu** [comment]
I have created https://github.com/datawire/apro/pull/3097 to check that this works with Ambassador Edge Stack.

Reviewers from Ambassador Labs: please to not approve this PR until you have verified that it works with Edge Stack.

**Pull request closed without merging**